src/System.Net.Sockets/src/System/Net/Sockets/OverlappedAsyncResult.Windows.cs

```diff
@@ -18,22 +18,21 @@ namespace System.Net.Sockets
     // from the BeginSend, BeginSendTo, BeginReceive, BeginReceiveFrom calls.
     internal partial class OverlappedAsyncResult : BaseOverlappedAsyncResult
     {
-        internal WSABuffer _singleBuffer;
-        internal WSABuffer[] _wsaBuffers;
-
-        internal IntPtr GetSocketAddressPtr()
+        internal unsafe byte* GetSocketAddressPtr()
         {
-            return Marshal.UnsafeAddrOfPinnedArrayElement(_socketAddress.Buffer, 0);
+            Debug.Assert(_socketAddress != null);
```

Comment:
Yes, good point.  The socketAddress.Buffer pinning code is a bit scary.  We should probably revisit it.